### PR TITLE
Add darc version parameter to allow quick version changes

### PR DESCRIPTION
**Pull request by dkurepa**
### To double check:
https://github.com/dotnet/arcade-services/issues/3798
* [ ] The right tests are in and the right validation has happened.  Guidance: https://github.com/dotnet/arcade/blob/main/Documentation/Validation.md

### CODE REVIEW

**Comment by dkurepa** [comment]
Test run: https://dev.azure.com/dnceng/internal/_build/results?buildId=2510604&view=results

File: eng/publishing/v3/validate-and-locate-build.ps1 [hunk 1]
```diff
@@ -1,11 +1,18 @@
 param(
   [Parameter(Mandatory = $true)][int] $BuildId,
-  [Parameter(Mandatory = $true)][string] $PromoteToChannelIds
+  [Parameter(Mandatory = $true)][string] $PromoteToChannelIds,
+  [Parameter(Mandatory = $true)][string] $DarcVersion
```
Comment (premun) [inline eng/publishing/v3/validate-and-locate-build.ps1:4]: Wait, why is this mandatory? I think we need to revert

Comment (premun) [inline eng/publishing/v3/validate-and-locate-build.ps1:4]: I think we might break people calling this without the version

Comment (premun) [reply eng/publishing/v3/validate-and-locate-build.ps1:4]: Only noticed now, sorry

Comment (premun) [reply eng/publishing/v3/validate-and-locate-build.ps1:4]: Ah, nevermind, there's no usage of this outside of arcade

Comment (dkurepa) [reply eng/publishing/v3/validate-and-locate-build.ps1:4]: yeah, it's not in eng/common

**Pull request merged**